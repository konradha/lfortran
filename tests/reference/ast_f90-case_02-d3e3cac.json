{
    "basename": "ast_f90-case_02-d3e3cac",
    "cmd": "lfortran --show-ast-f90 --no-color {infile}",
    "infile": "tests/../integration_tests/case_02.f90",
    "infile_hash": "7bbee224aea840614c0a2aef486a632a1120b0926e9f038ef87d47c1",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "ast_f90-case_02-d3e3cac.stdout",
    "stdout_hash": "6c97d7bece2c2c87354fc1ed1185a08b604d04ea96e5a953cedf2e1d",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}