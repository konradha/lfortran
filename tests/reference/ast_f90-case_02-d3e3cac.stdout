program case_02
implicit none

! local variable declaration
integer :: marks
marks = 81

select case (marks)

    case (91:100)
        print *, "Excellent!"

    case (81:90)
        print *, "Very good!"

    case (71:80)
        print *, "Well done!"

    case (61:70)
        print *, "Not bad!"

    case (41:60)
        print *, "You passed!"

    case (:40)
        print *, "Better try again!"

    case default
        print *, "Invalid marks"

end select
print *, "Your marks are ", marks

! case default at the top
select case (marks)

    case default
        print *, "Invalid marks"

    case (91:100)
        print *, "Excellent!"

    case (81:90)
        print *, "Very good!"

    case (71:80)
        print *, "Well done!"

    case (61:70)
        print *, "Not bad!"

    case (41:60)
        print *, "You passed!"

    case (:40)
        print *, "Better try again!"

end select
print *, "Your marks are ", marks

! case default in the middle
select case (marks)

    case (91:100)
        print *, "Excellent!"

    case (81:90)
        print *, "Very good!"

    case default
        print *, "Invalid marks"

    case (71:80)
        print *, "Well done!"

    case (61:70)
        print *, "Not bad!"

    case (41:60)
        print *, "You passed!"

    case (:40)
        print *, "Better try again!"

end select
print *, "Your marks are ", marks

end program case_02
